### PR TITLE
[ready-for-core-team-review]CRM-19804, added pre and post hook invokation for FinancialAccount

### PR DESCRIPTION
**Pull request by pradpnayak**
----------------------------------------
* CRM-19804: Pre() and Post() hook for Financial Account
  https://issues.civicrm.org/jira/browse/CRM-19804

### CODE REVIEW

File: CRM/Financial/BAO/FinancialAccount.php [hunk 1]
```diff
@@ -128,6 +136,14 @@ public static function add(&$params) {
       $financialAccount->opening_balance = $financialAccount->current_period_opening_balance = '0.00';
     }
     $financialAccount->save();
+
+    // invoke post hook
+    $op = 'create';
+    if (!empty($params['id'])) {
+      $op = 'edit';
+    }
```
Comment (monishdeb) [inline CRM/Financial/BAO/FinancialAccount.php]: @pradpnayak isn't this a repetition? I mean can we not initialise the $op parameter once ? and need optimisation 
```$op = empty($params['id']) ? 'create' : 'edit';```

Comment (pradpnayak) [inline CRM/Financial/BAO/FinancialAccount.php]: I think one can change the $params values in pre hook i.e they might unset($params['id']) when $op == 'edit' which will end up creating new entry in db rather update and hence post hook should invoke as $op == 'create'. 

 $op = empty($params['id']) ? 'create' : 'edit';
I was told by one of member from core team in past to use if () condition rather ternary operators. If this pattern is allowed i happy to optimize it and use it in future :)

Comment (monishdeb) [inline CRM/Financial/BAO/FinancialAccount.php]: ahh yes .. u're right

**Review by jitendrapurohit — APPROVED** [review]
tested, pre and post hook are working correctly.

**Pull request merged**